api/proto/info/info.proto

```diff
@@ -832,6 +832,19 @@ message ZInfoStorage {
   bool storageLocation = 4; // Storage location for app disks, images etc.
 }
 
+// Per snapshot information
+message ZInfoSnapshot {
+  // identifier for the snapshot, required
+  string id = 1;
+  // an app instance configuration, encoded as bytes, optional
+  // EVE needs to send this field in the cases when the snapshot was triggered
+  // locally and the Controller doesn't know that it has to save the app
+  // instance config at this moment.
```

Comment:
If we need a reference/identifier for the version of the configuration, then we should define what that reference looks like (is it the UUIDandVersion in the app instance config? something else like a sha or timestamp string generated by the controller each time it changes the app instance config?). 
If it is useful we can require that EVE-OS echo back such a reference string.

Reply:
Do we need to describe it in the API files? The only thing important here is that the IDs are unique. How EVE generates them - is a detail that should not be exposed on the API level.
If I'm wrong, I can provide the UUID info (see https://github.com/lf-edge/eve/pull/3018#discussion_r1099400415) here.

Reply:
EVE can't be trusted to generate UUIDs which will be unique across all of the devices and enterprises managed by a controller.

Reply:
As the snapshot IDs are handled/stored per VM, they need to be unique only per VM.
In the API messages, we pass the IDs only as a part of App Config / Info. So, it's ok.